web/modules/custom/dpl_paragraphs/dpl_paragraphs.module

```diff
@@ -223,7 +223,61 @@ function dpl_paragraphs_preprocess_paragraph__material_grid_automatic(array &$va
       'title' => $material_grid_title,
       'cql' => $cql,
       'selected-amount-of-materials-for-display' => $selected_amount_of_materials_for_display,
-      'button-text' => $button_text,
+      'button-text' => t('Show all'),
+      'blacklisted-search-branches-config' => $blacklistedSearchBranchesConfig,
+      'branches-config' => $branchesConfig,
+    ] + DplReactAppsController::externalApiBaseUrls(),
+  ];
+}
+
+/**
+ * Prepares the React app data for the Material Grid (Manual) paragraph.
+ *
+ * Data from the paragraph fields used to configure the Material Grid,
+ * includes the grid's title, and an array of materialId's (work-ids).
+ *
+ * @param array &$variables
+ *   Variables for the paragraph template, including the 'paragraph' entity.
+ *
+ * @see \Drupal\paragraphs\Entity\Paragraph
+ * @see dpl_react_apps/src/Controller/DplReactAppsController.php
+ */
+function dpl_paragraphs_preprocess_paragraph__material_grid_manual(array &$variables): void {
+
+  $paragraph = $variables['paragraph'] ?? NULL;
+
+  if (!($paragraph instanceof ParagraphInterface)  || !$paragraph->hasField('field_material_grid_title') || !$paragraph->hasField('field_work_id')) {
+    return;
+  }
+
+  // Extract field values.
+  $material_grid_title = $paragraph->get('field_material_grid_title')->getString();
+  $workIds = [];
+
+  // Check if the paragraph has the field and it contains at least one item.
+  if (!$paragraph->get('field_work_id')->isEmpty()) {
+
+    foreach ($paragraph->get('field_work_id') as $item) {
+      $workIds[] = $item->getString();
+    }
+  }
+
+  $json_encoded_workIds = json_encode($workIds);
+
+  // Build configurations for blacklisted search branches and all branches.
+  $branchSettings = \Drupal::service('dpl_library_agency.branch_settings');
+  $branchRepository = \Drupal::service('dpl_library_agency.branch.repository');
+  $blacklistedSearchBranchesConfig = DplReactAppsController::buildBranchesListProp($branchSettings->getExcludedSearchBranches());
+  $branchesConfig = DplReactAppsController::buildBranchesJsonProp($branchRepository->getBranches());
+
+  // Prepare the React app variables.
+  $variables['react_app_material_grid_manual'] = [
+    '#theme' => 'dpl_react_app',
+    '#name' => 'material-grid-manual',
+    '#data' => [
+      'material-ids' => $json_encoded_workIds,
```

Comment:
Try to remove the manual `json_encode`. As far as I remember our handling of data attributes for React apps will automatically JSON encode all complex values.